### ISSUE TITLE
Exploring the Potential of High-Quality Synthetic Datasets Using Translation-agent to Enhance the Performance of Large Multilingual Models

### ISSUE THREAD

**Issue by universea** [andrewyng/translation-agent#18]
**Background**
In the field of multilingual large models, especially for non-English corpora, there is often a problem of insufficient data quantity and poor quality. High-quality training data is crucial for model performance, particularly for under-resourced minority languages where a lack of sufficient training data is often a major limiting factor. Therefore, could translating an English corpus into various minority languages to create high-quality synthetic data corpora be a potential solution?

**Proposal**
The following steps might help generate and utilize synthetic datasets to enhance the capabilities of large multilingual models, especially for minority languages:

**Selection and Optimization of Translation Models**: Use the agent translation project or other efficient translation models as the translation tool to ensure high-quality output.

**Generation of Synthetic Datasets**: Utilize the aforementioned translation model to process English data, generating synthetic datasets in various target languages. Clean and validate the generated data to ensure its quality.

**Training and Fine-Tuning of Multilingual Large Language Models**: Use the synthetic datasets to train or fine-tune multilingual large models, observing improvements in the models' ability to process multilingual texts.

**Evaluation and Iteration**: Assess the model performance through standardized tests and real-world scenario testing, with a particular focus on the performance in minority languages, and iterate on the model and data generation processes based on feedback.

**Discussion**
This approach could greatly promote the generation of data and training of models for minority languages, as well as improve the overall performance and application breadth of multilingual models. I am eager to discuss the feasibility of this plan, particularly regarding the technical implementation and resource requirements. If you are interested in this topic or have relevant experience, please join the discussion and explore this field together.

**Comment by enismaxim1** [comment]
Having conducted research in a closely-related topic, I think that this approach is unlikely to work.

A useful mental model for how LLMs operate on different languages is the following: they first convert text (in any particular language) into a shared vector space representing _meaning_. Crucially, this vector space is language-agnostic: the LLM will convert a Russian text and an English text of equal meaning into a nearly-equal part of the vector space. The conversion from language into meaning is just translation, and the LLM will be lossy on this conversion on precisely the languages in which it is less effective at translating.

But the point is that the LLMs ability to effectively process low-resource language text is limited by its underlying translation ability. If we generate synthetic data, then we are limited by exactly the same thing, meaning that we are unlikely to be able to extract extra signal from synthetic data with this approach.

A related way to generate sytnethic data in a way that _does_ work is to use LLMs, which are better at translating particular documents than existing translation systems, in order to train the _smaller_ translation systems to match the translation performance of LLMs. If you're interested, see the issue #9 (or the paper https://arxiv.org/abs/2404.13813).

**Comment by random-yang** [comment]
> Having conducted research in a closely-related topic, I think that this approach is unlikely to work.Having conducted research in a closely-related topic, I think that this approach is unlikely to work.在对一个密切相关的主题进行研究后，我认为这种方法不太可能奏效。
> 
> A useful mental model for how LLMs operate on different languages is the following: they first convert text (in any particular language) into a shared vector space representing _meaning_. Crucially, this vector space is language-agnostic: the LLM will convert a Russian text and an English text of equal meaning into a nearly-equal part of the vector space. The conversion from language into meaning is just translation, and the LLM will be lossy on this conversion on precisely the languages in which it is less effective at translating.A useful mental model for how LLMs operate on different languages is the following: they first convert text (in any particular language) into a shared vector space representing _meaning_. Crucially, this vector space is language-agnostic: the LLM will convert a Russian text and an English text of equal meaning into a nearly-equal part of the vector space. The conversion from language into meaning is just translation, and the LLM will be lossy on this conversion on precisely the languages in which it is less effective at translating.LLM如何在不同语言上操作的一个有用的心智模型如下：它们首先将文本（在任何特定语言中）转换为表示_含义_的共享向量空间。至关重要的是，这个向量空间与语言无关：LLM将把意义相等的俄语文本和英语文本转换为向量空间中几乎相等的部分。从语言到意义的转换只是翻译，而LLM将在翻译效率较低的语言上失去这种转换。
> 
> But the point is that the LLMs ability to effectively process low-resource language text is limited by its underlying translation ability. If we generate synthetic data, then we are limited by exactly the same thing, meaning that we are unlikely to be able to extract extra signal from synthetic data with this approach.But the point is that the LLMs ability to effectively process low-resource language text is limited by its underlying translation ability. If we generate synthetic data, then we are limited by exactly the same thing, meaning that we are unlikely to be able to extract extra signal from synthetic data with this approach.但关键是LLM有效处理低资源语言文本的能力受到其底层翻译能力的限制。如果我们生成合成数据，那么我们受到完全相同的东西的限制，这意味着我们不太可能用这种方法从合成数据中提取额外的信号。
> 
> A related way to generate sytnethic data in a way that _does_ work is to use LLMs, which are better at translating particular documents than existing translation systems, in order to train the _smaller_ translation systems to match the translation performance of LLMs. If you're interested, see the issue #9 (or the paper https://arxiv.org/abs/2404.13813).A related way to generate sytnethic data in a way that _does_ work is to use LLMs, which are better at translating particular documents than existing translation systems, in order to train the _smaller_ translation systems to match the translation performance of LLMs. If you're interested, see the issue #9 (or the paper https://arxiv.org/abs/2404.13813).以_确实_工作的方式生成同步数据的一种相关方法是使用LLM，它比现有翻译系统更擅长翻译特定文档，以便训练_较小的_翻译系统以匹配LLM的翻译性能。如果您感兴趣，请参阅问题#9（或论文https://arxiv.org/abs/2404.13813）。

You mean that even for LLM, the low-resource language knowledge it contains is very little, resulting in our inability to effectively distill this knowledge for further improvement of low-resource language translation?

**Comment by random-yang** [comment]
> Having conducted research in a closely-related topic, I think that this approach is unlikely to work.Having conducted research in a closely-related topic, I think that this approach is unlikely to work.在对一个密切相关的主题进行研究后，我认为这种方法不太可能奏效。
> 
> A useful mental model for how LLMs operate on different languages is the following: they first convert text (in any particular language) into a shared vector space representing _meaning_. Crucially, this vector space is language-agnostic: the LLM will convert a Russian text and an English text of equal meaning into a nearly-equal part of the vector space. The conversion from language into meaning is just translation, and the LLM will be lossy on this conversion on precisely the languages in which it is less effective at translating.A useful mental model for how LLMs operate on different languages is the following: they first convert text (in any particular language) into a shared vector space representing _meaning_. Crucially, this vector space is language-agnostic: the LLM will convert a Russian text and an English text of equal meaning into a nearly-equal part of the vector space. The conversion from language into meaning is just translation, and the LLM will be lossy on this conversion on precisely the languages in which it is less effective at translating.LLM如何在不同语言上操作的一个有用的心智模型如下：它们首先将文本（在任何特定语言中）转换为表示_含义_的共享向量空间。至关重要的是，这个向量空间与语言无关：LLM将把意义相等的俄语文本和英语文本转换为向量空间中几乎相等的部分。从语言到意义的转换只是翻译，而LLM将在翻译效率较低的语言上失去这种转换。
> 
> But the point is that the LLMs ability to effectively process low-resource language text is limited by its underlying translation ability. If we generate synthetic data, then we are limited by exactly the same thing, meaning that we are unlikely to be able to extract extra signal from synthetic data with this approach.But the point is that the LLMs ability to effectively process low-resource language text is limited by its underlying translation ability. If we generate synthetic data, then we are limited by exactly the same thing, meaning that we are unlikely to be able to extract extra signal from synthetic data with this approach.但关键是LLM有效处理低资源语言文本的能力受到其底层翻译能力的限制。如果我们生成合成数据，那么我们受到完全相同的东西的限制，这意味着我们不太可能用这种方法从合成数据中提取额外的信号。
> 
> A related way to generate sytnethic data in a way that _does_ work is to use LLMs, which are better at translating particular documents than existing translation systems, in order to train the _smaller_ translation systems to match the translation performance of LLMs. If you're interested, see the issue #9 (or the paper https://arxiv.org/abs/2404.13813).A related way to generate sytnethic data in a way that _does_ work is to use LLMs, which are better at translating particular documents than existing translation systems, in order to train the _smaller_ translation systems to match the translation performance of LLMs. If you're interested, see the issue #9 (or the paper https://arxiv.org/abs/2404.13813).以_确实_工作的方式生成同步数据的一种相关方法是使用LLM，它比现有翻译系统更擅长翻译特定文档，以便训练_较小的_翻译系统以匹配LLM的翻译性能。如果您感兴趣，请参阅问题#9（或论文https://arxiv.org/abs/2404.13813）。

You mean that even for LLM, the low-resource language knowledge it contains is very little, resulting in our inability to effectively distill this knowledge for further improvement of low-resource language translation?

> Having conducted research in a closely-related topic, I think that this approach is unlikely to work.
> 
> A useful mental model for how LLMs operate on different languages is the following: they first convert text (in any particular language) into a shared vector space representing _meaning_. Crucially, this vector space is language-agnostic: the LLM will convert a Russian text and an English text of equal meaning into a nearly-equal part of the vector space. The conversion from language into meaning is just translation, and the LLM will be lossy on this conversion on precisely the languages in which it is less effective at translating.
> 
> But the point is that the LLMs ability to effectively process low-resource language text is limited by its underlying translation ability. If we generate synthetic data, then we are limited by exactly the same thing, meaning that we are unlikely to be able to extract extra signal from synthetic data with this approach.
> 
> A related way to generate sytnethic data in a way that _does_ work is to use LLMs, which are better at translating particular documents than existing translation systems, in order to train the _smaller_ translation systems to match the translation performance of LLMs. If you're interested, see the issue #9 (or the paper https://arxiv.org/abs/2404.13813).

You mean that even for LLM, the low-resource language knowledge it contains is very little, resulting in our inability to effectively distill this knowledge for further improvement of low-resource language translation?

**Comment by random-yang** [comment]
> Having conducted research in a closely-related topic, I think that this approach is unlikely to work.
> 
> A useful mental model for how LLMs operate on different languages is the following: they first convert text (in any particular language) into a shared vector space representing _meaning_. Crucially, this vector space is language-agnostic: the LLM will convert a Russian text and an English text of equal meaning into a nearly-equal part of the vector space. The conversion from language into meaning is just translation, and the LLM will be lossy on this conversion on precisely the languages in which it is less effective at translating.
> 
> But the point is that the LLMs ability to effectively process low-resource language text is limited by its underlying translation ability. If we generate synthetic data, then we are limited by exactly the same thing, meaning that we are unlikely to be able to extract extra signal from synthetic data with this approach.
> 
> A related way to generate sytnethic data in a way that _does_ work is to use LLMs, which are better at translating particular documents than existing translation systems, in order to train the _smaller_ translation systems to match the translation performance of LLMs. If you're interested, see the issue #9 (or the paper https://arxiv.org/abs/2404.13813).

You mean that even for LLM, the low-resource language knowledge it contains is very little, resulting in our inability to effectively distill this knowledge for further improvement of low-resource language translation?

**Comment by enismaxim1** [comment]
No, I just mean that translating English data in a low-resource language using an LLM and then training it on that same data is unlikely to yield any performance improvement. The argument is that the LLMs level of multilinguality should come from its ability to translate therefore translation does not provide any _additional_ useful data to the LLM.

**Comment by universea** [comment]
> No, I just mean that translating English data to a low-resource language using an LLM and then training it on that same data is unlikely to yield any performance improvement. The argument is that the LLMs level of multilinguality should come from its ability to translate therefore translation does not provide any _additional_ useful data to the LLM.

Thank you for your comment. And, I want to know if this method can improve the reasoning, logic, and mathematical abilities of large models in low-resource languages.

**Comment by enismaxim1** [comment]
I would guess that you cannot. Intuitively I would expect that reasoning/logic/math capabilities should be independent of any particular language (instead operating on the shared latent space of meaning). Therefore I would expect that the logic/math/reasoning capabilities should again be bottlenecked by translation/language understanding, which I don't expect this method can improve.

**Comment by universea** [comment]
I mostly support your viewpoints, but I have also noticed a phenomenon. For example, many large models can answer an English math question quite well, but when the same math question is presented in other languages, the accuracy and quality of the answers from the large model tend to decline. If we translate the math question from a lesser-known language into English, the quality and accuracy of the model's answers improve. How do you view this issue?

**Comment by enismaxim1** [comment]
> I mostly support your viewpoints, but I have also noticed a phenomenon. For example, many large models can answer an English math question quite well, but when the same math question is presented in other languages, the accuracy and quality of the answers from the large model tend to decline. If we translate the math question from a lesser-known language into English, the quality and accuracy of the model's answers improve. How do you view this issue?

This is interesting. If it is true, then these methods should work. Did you translate the problem into English using an LLM or with something else (like Google)?
